### PR TITLE
treewide: use fetchFromGitHub when possible

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
This is just 7 out of possibly hundreds that ripgrep found, I may need to find a way to automate this since it's quite time consuming. Found these via

```ShellSession
$ rg -l -U -P 'fetchurl[^]]*\n*[^]]*?\bgithub'
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/wiki-js/default.nix [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
   version = "2.5.201";
 
-  src = fetchurl {
-    url = "https://github.com/Requarks/wiki/releases/download/${version}/${pname}.tar.gz";
-    sha256 = "sha256-k2G+jUne/lq0dRJsIQpWlRFg1Nq92bM28YkawKOKlsI=";
+  src = fetchFromGitHub {
+    owner = "Requarks";
+    repo = "wiki";
+    rev = version;
+    sha256 = "pM2sbs6ruzXxg1EiWTzPgj8IORA2Kawe79POlrQFSkc=";
```
Comment (Ma27) [inline pkgs/servers/web-apps/wiki-js/default.nix:11]: I'm pretty sure that this one is wrong since it's a tarball which also contains node_modules.

**Comment by rnhmjoj** [comment]
@grahamcOfBorg build diod

**Comment by legendofmiracles** [comment]
> I may need to find a way to automate this
How would one do this? 
I know of another thing that could need a treewide change

**Comment by siraben** [comment]
@legendofmiracles please elaborate what treewide you would like to see.

**Comment by siraben** [comment]
Reverting change to diod.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 122582` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>axel</li>
    <li>bin_replace_string</li>
    <li>check-wmiplus</li>
    <li>duply</li>
    <li>matterbridge</li>
    <li>mdbtools</li>
    <li>peek</li>
    <li>stockfish</li>
    <li>txt2man</li>
    <li>uchess</li>
  </ul>
</details>

**Comment by siraben** [comment]
Superceded by #163582

**Pull request closed without merging**